src/convert-data.jl

```diff
@@ -39,12 +68,28 @@ function sexp(v::DataArray)
     rv
 end
 
-## PooledDataArray to sexp conversion.
-function sexp{T<:Compat.String,R<:Integer}(v::PooledDataArray{T,R})
-    rv = sexp(v.refs)
-    setattrib!(rv, Const.LevelsSymbol, sexp(v.pool))
-    setattrib!(rv, Const.ClassSymbol, sexp("factor"))
-    rv
+## CategoricalArray to sexp conversion.
+for typ in [:NullableCategoricalArray, :CategoricalArray]
+    @eval begin
+        function sexp{T<:Compat.String,N,R<:Integer}(v::$typ{T,N,R})
+            rv = protect(sexp(v.refs))
+            try
+                for (i,isna) = enumerate(v.refs .== 0)
+                    if isna
+                        rv[i] = naeltype(eltype(rv))
+                    end
+                end
+                setattrib!(rv, Const.LevelsSymbol, sexp(v.pool.index))
```

Comment:
Ah, indeed, at first I didn't intend it to be used outside of the package. For now the best approach is `CategoricalArrays.index(v.pool)`. I'll try to see if I can improve the API in next releases.